test/gtest/ucs/test_conn_match.cc

```diff
@@ -76,32 +72,40 @@ class test_conn_match : public ucs::test {
         ucs_conn_match_cleanup(&m_conn_match_ctx);
     }
 
-    void generate_random_address(std::string &address, size_t length) {
+    void *generate_random_address(size_t length) {
         std::string uuid_str = ucs::to_string(ucs_generate_uuid(0));
         size_t string_length = length - uuid_str.size();
+        void *address        = new uint8_t[length];
 
-        address.clear();
+        if (address == NULL) {
+            UCS_TEST_ABORT("failed to allocate memory for address");
+        }
 
         if (string_length) {
-            ucs::generate_random_str(address, string_length);
+            std::string adress_string;
+            ucs::generate_random_str(adress_string, string_length);
+            memcpy(address, adress_string.c_str(), string_length);
```

Comment:
why need both str and uuid? generate_random_str should be enough
it can use uuid as seed

Reply:
the random string could be duplicated and then the same address will be used several times

Reply:
do we want the same address will be used several times?

Reply:
I think we need to test random insert/retrieve different addresses in random order, then we could remove generating uuid
I'll do it

Reply:
done